### PR TITLE
A0-2232: Truncate long address with ellipsis

### PR DESCRIPTION
**Pull request by jalooc**
image

### CODE REVIEW

File: packages/extension-ui/src/components/FaviconBox.tsx [hunk 1]
```diff
@@ -56,12 +56,16 @@ export default styled(FaviconBox)(
...
+  
+  min-width: 0; /* To enable ellipsis for flex grandchildren (https://css-tricks.com/flexbox-truncated-text/) */
 
   .favicon-box-content {
     display: flex;
     flex-direction: row;
     align-items: center;
     gap: 8px;
+
+    min-width: 0; /* To enable ellipsis for flex grandchildren (https://css-tricks.com/flexbox-truncated-text/) */
```
Comment (jalooc) [inline packages/extension-ui/src/components/FaviconBox.tsx:68]: Both levels (= all levels) of descendants of the flex ancestor need `min-width: 0` for it to work.

File: packages/extension-ui/src/components/RemoveAuth.tsx [hunk 2]
```diff
@@ -20,6 +20,7 @@ interface Props {
 const StyledFaviconBox = styled(FaviconBox)`
   box-sizing: border-box;
   margin: 0px auto;
+  max-width: calc(100% - 16px);
```
Comment (jalooc) [inline packages/extension-ui/src/components/RemoveAuth.tsx:23]: Without this, the outter box would extend to the edges of the window.

File: packages/extension-ui/src/components/FaviconBox.tsx [hunk 3]
```diff
@@ -56,12 +56,16 @@ export default styled(FaviconBox)(
   border-radius: 8px;
   transition: 0.2s ease;
   justify-content: space-evenly;
+  
+  min-width: 0; /* To enable ellipsis for flex grandchildren (https://css-tricks.com/flexbox-truncated-text/) */
```
Comment (youPickItUp) [inline packages/extension-ui/src/components/FaviconBox.tsx:60]: I think this one isn't required

File: packages/extension-ui/src/components/RemoveAuth.tsx (hunk 2 above)
Comment (youPickItUp) [reply packages/extension-ui/src/components/RemoveAuth.tsx:23]: I think `max-width: calc(100% - 24px);` would look better. On the address below, apart from same trick, there is also 4px inline padding:

image

**Review by youPickItUp — CHANGES_REQUESTED** [review]

File: packages/extension-ui/src/components/FaviconBox.tsx (hunk 3 above)
Comment (jalooc) [reply packages/extension-ui/src/components/FaviconBox.tsx:60]: Doesn't work without it. Also I think it makes sense for it to be required - this element has a flex parent on its own, so needs `min-width` reset as well.

Comment (youPickItUp) [reply packages/extension-ui/src/components/FaviconBox.tsx:60]: I'm sorry to bother you with such a nitpick, but I tested on chrome and firefox and it works :shrug: .
Perhaps that's a Brave thing, but would you be as kind as to check again :sweat_smile:

**Review by youPickItUp — APPROVED** [review]

File: packages/extension-ui/src/components/FaviconBox.tsx (hunk 3 above)
Comment (jalooc) [reply packages/extension-ui/src/components/FaviconBox.tsx:60]: Huh, you are right - I guess when checking it previously, I was mistakenly validating for `max-width` instead 🤦 Works as you're saying on both chrome and brave. Kudos for persistence 😄

**Pull request merged**